tests/data_generation/helper/induce_causal_effect.py

```
@@ -0,0 +1,331 @@
+        if not parquet_files:
+            raise ValueError(f"No parquet files found in {shard_dir}")
+
+        dfs, shards = [], {}
+        for i, filename in enumerate(parquet_files):
+            file_path = os.path.join(shard_dir, filename)
+            try:
+                shard = pd.read_parquet(file_path)
+                shards[i] = shard.subject_id.unique().tolist()
+                dfs.append(shard)
+            except Exception as e:
+                raise ValueError(f"Error reading {file_path}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)